Marlin/src/lcd/e3v2/proui/endstop_diag.cpp

```diff
@@ -57,11 +57,12 @@ void draw_es_label(FSTR_P const flabel=nullptr) {
   DWINUI::moveBy(0, 25);
 }
 
-void draw_es_state(const bool is_hit) {
+void draw_es_state(const bool is_hit, const bool is_fil) {
   const uint8_t LM = 130;
   DWINUI::cursor.x = LM;
-  dwinDrawRectangle(1, hmiData.colorPopupBg, LM, DWINUI::cursor.y, LM + 100, DWINUI::cursor.y + 20);
-  is_hit ? DWINUI::drawString(RGB(31,31,16), F(STR_ENDSTOP_HIT)) : DWINUI::drawString(RGB(16,63,16), F(STR_ENDSTOP_OPEN));
+  dwinDrawRectangle(1, hmiData.colorPopupBg, LM, DWINUI::cursor.y, LM + 120, DWINUI::cursor.y + 20);
+  is_fil ? (is_hit ? DWINUI::drawString(RGB(16, 63, 16), F("PRESENT")) : DWINUI::drawString(RGB(31, 31, 16), F("Runout Detected"))) :
+  (is_hit ? DWINUI::drawString(RGB(31, 31, 16), F(STR_ENDSTOP_HIT)) : DWINUI::drawString(RGB(16, 63, 16), F(STR_ENDSTOP_OPEN)));
```

Comment:
If these are for display on the LCD to common users they should use translatable strings with `GET_TEXT_F` rather than inline strings or `STR_*` strings that are intended for serial output.

Reply:
other than making up new ones, I'm looking through what is available, what about `MSG_YES` for hit, and `MSG_NO` for open?

as for "PRESENT" => `MSG_FILAMENT` and "Runout Detected" => `MSG_RUNOUT_SENSOR`, or it could be `MSG_FILAMENT_CHANGE_HEADER_LOAD` ("LOAD FILAMENT")

Reply:
Making up new ones is the correct way to go for strings not yet defined. Prefer pre-existing ones where possible. I went ahead and added a commit to internationalize various strings. If you find any more that need translation as you go along, please add them.

Reply:
im going to leave the `STR_*` for now because a. I can't find a suitable `MSG_` replacement, and b. im sure "TRIGGERED" in big red letters is universally understood 😆.